### ISSUE TITLE
32-character UUID string can not be decoded as UUID

### ISSUE THREAD

**Issue by cmanallen** [jcrist/msgspec#388]
### Description

These UUIDs can be produced by calling `import uuid; uuid.uuid4().hex`.

```python
>>> import msgspec
>>> import uuid
>>>
>>> msgspec.json.decode(b'"f3eed2ee-eafa-417e-9de6-c2d2fe630122"', type=uuid.UUID)
UUID('f3eed2ee-eafa-417e-9de6-c2d2fe630122')
>>>
>>> msgspec.json.decode(b'"f3eed2eeeafa417e9de6c2d2fe630122"', type=uuid.UUID)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
msgspec.ValidationError: Invalid UUID
```

**Comment by jcrist** [comment]
We currently strictly follow the ABNF grammar defined in https://www.rfc-editor.org/rfc/rfc4122.html#section-3. This requires hyphens as separators.

Is there precedence for systems supporting alternative representations? Are there downsides to being more flexible here? A quick survey shows that golang only supports the hyphenated format, while rust's uuid with serde supports a couple common forms.

**Comment by jcrist** [comment]
Makes sense to me! This has been fixed in #392. If you want to try it out before the next release you can [install from github](https://jcristharif.com/msgspec/install.html#installing-from-github).

**Comment by cmanallen** [comment]
@jcrist Awesome!  Loving msgspec btw.  I've incorporated into two personal projects.  Thank you for this!
